services/src/main/java/org/fao/geonet/api/users/UsersApi.java

```diff
@@ -85,6 +86,11 @@
     description = "User operations")
 @Controller("users")
 public class UsersApi {
+    /**
+     * Username pattern with allowed chars. Username may only contain alphanumeric characters or single hyphens,
+     * single at signs or single dots. Cannot begin or end with a hyphen, at sign or dot.
+     */
+    private static final String USERNAME_PATTERN = "^[a-zA-Z0-9]+([-_.@]?[a-zA-Z0-9]+)*$";
```

Comment:
What is the purpose of only allowing one -_. in the username? Usernames are generally email addresses in a lot of cases so I understand the "@" being limited to one occurrence.  But the -_. may occur multiple times.

I'm quite sure the following are valid email addresses which contain multiple "." and "-"
user@mail.example.com 
user@sample-hyphen-subdns.example.com

There are also some domains mentioned here that have multiple "_" however I have rarely seen this.
https://stackoverflow.com/questions/2180465/can-domain-name-subdomains-have-an-underscore-in-it

Reply:
@ianwallen I don't think the intention of the change was to support only 1 `-`, `_` or `.`.

I think the regular expression should be like this:

`^([a-zA-Z0-9]+)([a-zA-Z0-9-_.@]+[a-zA-Z0-9]+)*$`

Reply:
I agree that the regular expression that you have supplied makes more sense.

Reply:
@ianwallen please check the latest commit: https://github.com/geonetwork/core-geonetwork/pull/6001/commits/a5cc17522a94f24fd7970d61972d7d3261ae62fa